WireDomain/Tests/WireDomainTests/Event Processing/UserEventProcessor/UserPushRemoveEventProcessorTests.swift

```diff
@@ -0,0 +1,89 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireAPI
+import WireAPISupport
+import WireDataModel
+import WireDataModelSupport
+import XCTest
+
+@testable import WireDomain
+
+final class UserPushRemoveEventProcessorTests: XCTestCase {
+
+    var sut: UserPushRemoveEventProcessor!
+
+    var coreDataStack: CoreDataStack!
+    let coreDataStackHelper = CoreDataStackHelper()
+    let modelHelper = ModelHelper()
+
+    var context: NSManagedObjectContext {
+        coreDataStack.syncContext
+    }
+
+    override func setUp() async throws {
+        try await super.setUp()
+        coreDataStack = try await coreDataStackHelper.createStack()
+        sut = UserPushRemoveEventProcessor(
+            repository: UserRepository(
+                context: context,
+                usersAPI: MockUsersAPI()
+            )
+        )
+    }
+
+    override func tearDown() async throws {
+        try await super.tearDown()
+        coreDataStack = nil
+        sut = nil
+        try coreDataStackHelper.cleanupDirectory()
+    }
+
+    // MARK: - Tests
+
+    func testProcessEvent_It_Removes_Push_Token_From_Defaults() async throws {
+        // Given
+
+        let defaults = UserDefaults.standard
```

Comment:
Perhaps we should use a temporary defaults to avoid polluting the real one.

Reply:
done in https://github.com/wireapp/wire-ios/pull/1948/commits/09836061f1870849edc69be3d204a5c2afb69df8